src/components/section/NewEventCard.tsx

```diff
@@ -0,0 +1,198 @@
+        <label>Time zone</label>
+        <RoundedTimezoneInput
+          className="px-0 text-sm"
+          value={timeZoneTag}
+          onChange={(newTimeZoneTag) => {
+            setStartTime(moveTime(startTime, timeZoneTag, newTimeZoneTag));
+            setEndTime(moveTime(endTime, timeZoneTag, newTimeZoneTag));
+            setTimeZoneTag(newTimeZoneTag);
+          }}
+        />
+      </div>
+      <div className="input-field text-sm">
+        <label>Select days</label>
+        <RoundedListbox
+          className="px-0 text-sm"
+          options={selectDaysOptions}
+          value={selectDaysType}
+          onChange={(value) => {
+            setSelectedDays([]);
+            setSelectDaysType(value);
+          }}
+        />
+      </div>
+      {selectDaysType === "DAYSOFWEEK" && (
+        <>
+          <div className="input-field">
+            <label>Days of week</label>
+            <text>Click or drag on days to select</text>
+          </div>
+          <DateSelect week value={selectedDays} onChange={setSelectedDays} />
+        </>
+      )}
+      {selectDaysType === "DATES" && (
+        <>
+          <div className="input-field">
+            <label>Dates</label>
+            <text>Click or drag on dates to select</text>
+          </div>
+          <DateSelect value={selectedDays} onChange={setSelectedDays} />
+        </>
+      )}
+      <div className="input-field">
+        <label>Timespan</label>
+        <TimespanSelector
+          timeZone={timeZoneTag}
+          start={startTime}
+          end={endTime}
+          onChangeStart={setStartTime}
+          onChangeEnd={setEndTime}
+        />
+      </div>
+      <ButtonWithState
+        className="primary-button mt-3 py-3 text-sm"
+        loadingClassName="primary-button-loading mt-3 py-3 text-sm"
+        disabledClassName="rounded-button-disabled mt-3 py-3 text-sm"
+        loading={creatingEvent}
+        disabled={!canSubmit}
+        onClick={() => void handleCreateEvent()}
+      >
+        Create Event
+      </ButtonWithState>
+      <div className="text-center text-xs text-gray-400">
+        Timezone, days and time span cannot be <br />
+        changed after the event is created
+      </div>
+    </>
+  );
+};
+
+const JoinExistingEventSection = () => {
+  const router = useRouter();
+  const [joinCode, setJoinCode] = useState("");
+  const { data: event, isLoading } = useEventWithJoinCode(joinCode);
+
+  const handleJoinEvent = () => {
+    const eventId = event?.id;
+    if (eventId) {
+      void router.push(`/event/${eventId}`);
+    }
+  };
+
+  return (
+    <>
+      <div className="input-field">
+        <label>Event code</label>
+        <text>Ask the host to provide the 6-digit event code</text>
+        <input
+          className="rounded-input"
+          placeholder="xxxxxx"
+          maxLength={6}
+          value={joinCode}
+          onChange={(e) => setJoinCode(e.target.value)}
+        />
+      </div>
```

Comment:
Can we debounce this input so we don't query by each letter input?

Reply:
Also, I believe I once set this input to be number-only, sth like this:

```jsx
<input type="number" />
```
This way we can ensure ppl only inputs numbers in the first place, rather than not enabling the button if it's not number. Seems like a better UX to me

Reply:
Setting the input type to a number will interrupt the maxLength property.

Also, this is not recommended by MDN documentation ("The number input type is not appropriate for values that happen to only consist of numbers but aren't strictly speaking a number, such as postal codes in many countries or credit card numbers." from [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number)). The suggested way does not yield visible results in browsers (I can still input characters), so I would recommend simply omitting this setting.

Working on debouncing.

Reply:
I may, however, try to manually examine the input and reject invalid ones by not updating the value.

Reply:
Now the behavior of the input is as follows:

1. It only accepts digit characters with a maximum length of 6;
2. It only queries the database when the length of input reaches 6;
3. The query is debounced by 100ms.